### PR TITLE
drivers: sensor: Add ADXL362 interrupt handling

### PR DESCRIPTION
**Pull request by hmhalvorsen**
The ADXL362 driver is expanded to support interrupt handling.

Note: This PR builds upon a previous [PR for ADXL362,](https://github.com/zephyrproject-rtos/zephyr/pull/14030) yet to be merged.

### CODE REVIEW

**Comment by hmhalvorsen** [comment]
PR updated. @MaureenHelm please review.

File: drivers/sensor/adxl362/Kconfig [hunk 1]
```diff
@@ -58,4 +58,56 @@ config ADXL362_ACCEL_ODR_400
...
+
+config ADXL362_TRIGGER
+	bool
+
+config ADXL362_THREAD_PRIORITY
+	int "Thread priority"
+	depends on ADXL362_TRIGGER_OWN_THREAD && ADXL362_TRIGGER
+	default 10
+	help
+		Priority of thread used by the driver to handle interrupts.
```
Comment (MaureenHelm) [inline drivers/sensor/adxl362/Kconfig]: Please fix the indentation. It should be two spaces after help

File: drivers/sensor/adxl362/adxl362.c [hunk 2]
```diff
@@ -69,6 +69,33 @@ static inline int adxl362_set_reg(struct device *dev, u16_t register_value,
 				  count);
 }
 
+int adxl362_reg_write_mask(struct device *dev, u8_t register_address,
+			   u32_t mask, u8_t data)
```
Comment (MaureenHelm) [inline drivers/sensor/adxl362/adxl362.c]: why is the mask not `u8_t`?

File: drivers/sensor/adxl362/adxl362.c [hunk 3]
```diff
@@ -80,6 +107,37 @@ static inline int adxl362_get_reg(struct device *dev, u8_t *read_buf,
...
+				  1);
+
+	if (ret) {
+		return ret;
+	}
+
+	return ret = adxl362_reg_access(adxl362_data,
+					ADXL362_WRITE_REG,
+					ADXL362_REG_INTMAP2,
+					&int1,
```
Comment (MaureenHelm) [inline drivers/sensor/adxl362/adxl362.c]: Should this be `&int2`?

File: drivers/sensor/adxl362/adxl362.c [hunk 4]
```diff
@@ -248,16 +306,54 @@ static int axl362_acc_config(struct device *dev, enum sensor_channel chan,
...
+			    enum sensor_attribute attr,
+			    const struct sensor_value *val)
+{
+	u8_t reg;
+	u16_t threshold = val->val1;
+	size_t ret;
+
+	if (!(SENSOR_CHAN_ACCEL_X ||
+	      SENSOR_CHAN_ACCEL_Y ||
+	      SENSOR_CHAN_ACCEL_Z)) {
```
Comment (MaureenHelm) [inline drivers/sensor/adxl362/adxl362.c]: I think you mean to compare `chan` to these values?

File: drivers/sensor/adxl362/adxl362_trigger.c [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+#include <kernel.h>
+#include <sensor.h>
+
+#include "adxl362.h"
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_DECLARE(ADXL362);
+
+static struct k_mutex trigger_mutex;
```
Comment (MaureenHelm) [inline drivers/sensor/adxl362/adxl362_trigger.c]: This should be in `struct adxl362_data`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this sensor to `tests/drivers/build_all/sensors_trigger_a_h.conf`

File: drivers/sensor/adxl362/adxl362_trigger.c (hunk 5 above)
Comment (hmhalvorsen) [reply drivers/sensor/adxl362/adxl362_trigger.c]: Right. It is now moved.

File: drivers/sensor/adxl362/adxl362.c (hunk 4 above)
Comment (hmhalvorsen) [reply drivers/sensor/adxl362/adxl362.c]: Oh yes, of course. Thanks!

File: drivers/sensor/adxl362/adxl362.c (hunk 3 above)
Comment (hmhalvorsen) [reply drivers/sensor/adxl362/adxl362.c]: Yes, this is a mistake.

File: drivers/sensor/adxl362/adxl362.c (hunk 2 above)
Comment (hmhalvorsen) [reply drivers/sensor/adxl362/adxl362.c]: Changed.

File: drivers/sensor/adxl362/Kconfig (hunk 1 above)
Comment (hmhalvorsen) [reply drivers/sensor/adxl362/Kconfig]: Fixed.

**Comment by hmhalvorsen** [comment]
This PR will be updated shortly. Please do not merge until update is live.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**